hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/Sinks.java

```diff
@@ -109,7 +111,52 @@ private Sinks() {
      */
     @Nonnull
     public static <K, V> Sink<Entry<K, V>> map(@Nonnull String mapName) {
-        return new SinkImpl<>("mapSink(" + mapName + ')', writeMapP(mapName), false, entryKey());
+        return map(mapName, Entry::getKey, Entry::getValue);
+    }
+
+    /**
+     * Returns a sink that converts the key and value of {@code Map.Entry}s it
+     * receives to {@link HazelcastJsonValue} and put them into a Hazelcast
+     * {@code IMap} with the specified name.
+     * <p>
+     * This sink provides the exactly-once guarantee thanks to <i>idempotent
+     * updates</i>. It means that the value with the same key is not appended,
+     * but overwritten. After the job is restarted from snapshot, duplicate
+     * items will not change the state in the target map.
+     * <p>
+     * The default local parallelism for this sink is 1.
+     *
+     * @since 4.1
+     */
+    @Nonnull
+    public static <K, V> Sink<Entry<K, V>> map(@Nonnull String mapName, boolean jsonKey, boolean jsonValue) {
```

Comment:
I don't think this overload adds much, and is a bit confusing. It should be enough to provide the function reference ? i.e. `JsonUtil::asJson` or something so I would just remove this variant and mention it in javadoc